### PR TITLE
Feat/minimax m2.5 support

### PR DESCRIPTION
**Pull request by xs1997zju**
## Summary

Add full integration for **MiniMax-M2.5** (256 experts, top-8 routing), including:

- **Model spec plugin** (`slime_plugins/models/minimax_m2.py`): Custom `SelfAttention` with full-dimension QK Norm (RMSNorm over all heads concatenated, with TP gather/scatter)
- **mbridge weight bridge** (`slime_plugins/mbridge/minimax_m2.py`): HF ↔ Megatron weight mapping extending `Qwen2MoEBridge`
- **Megatron-to-HF converter** (`slime/backends/megatron_utils/megatron_to_hf/minimax_m2.py`): Reverse conversion for saving trained checkpoints back to HF format
- **Shell scripts** (`scripts/`): Model architecture args, RL training launch script, and 3-script HF ↔ Megatron weight conversion pipeline

### Key architecture differences from standard Qwen2MoE

| Feature | MiniMax-M2.5 | Qwen2MoE |
|---|---|---|
| MoE prefix | `block_sparse_moe` (w1/w2/w3) | `mlp` |
| QK Norm | Full-dimension (all heads concat) | Per-head |
| Router | Sigmoid + `e_score_correction_bias` | Softmax |
| RoPE | Partial (50%) | Full |
| Experts | 256 × 62 layers | varies |

### CODE REVIEW

**Review by zhuzilin — APPROVED** [review]

**Comment by zhuzilin** [comment]
Thank you for the PR! Just a quick check — could you provide some evidence that the implementation is working correctly, such as W&B screenshots, logs, or validation outputs?

**Comment by xs1997zju** [comment]
> Thank you for the PR! Just a quick check — could you provide some evidence that the implementation is working correctly, such as W&B screenshots, logs, or validation outputs?
@zhuzilin 
image
aime eval results with dapo-math-17k train
 parameters:

`ROLLOUT_ARGS=(
   --prompt-data dapo-math-17k.jsonl
   --input-key prompt
   --label-key label
   --apply-chat-template
   --rollout-shuffle
   --rm-type deepscaler
   --num-rollout 3000
   --rollout-batch-size 32
   --n-samples-per-prompt 8
   --rollout-max-response-len 16384
   --rollout-temperature 1
   --global-batch-size 256
)

EVAL_ARGS=(
   --eval-interval 20
   --eval-prompt-data aime-2024.jsonl
   --n-samples-per-eval-prompt 4
   --eval-max-response-len 16384
   --eval-top-p 1
)`

**Comment by Lynnzake** [comment]
Hello, Any Plan to support minimax m2.7?
